### PR TITLE
Ux enhancements

### PR DESCRIPTION
**Pull request by nikokreissl**
- Added quick edit functionality for challenges on `objective detail`
- Changed `game edit / create` to be done with buttons instead of input
- Changed `input to range` on tactic general instructions edit
- Added `descriptions to pages` to make clear what it is about
- Implemented `toasts with react/toastify`
- Added components for links and buttons and updated them to be semantically correct (links > href / button > function)
- removed main from individual pages and added to _app.js

Most of the files changed in this PR are coming from updating buttons and links as well as removing main from the individual pages

### CODE REVIEW

File: components/Competition/CreateCompetition/index.js [hunk 1]
```diff
@@ -26,42 +36,58 @@ export default function CreateCompetitionForm() {
     setCompetitionGameInput(Number(event.target.value));
   }
 
-  function handleSubmit(event) {
-    event.preventDefault();
```
Comment (actyralf) [inline components/Competition/CreateCompetition/index.js:30]: Why did you delete `event.preventDefault()`? Your app will be reloaded when you don't suppress the default behavior.

File: components/Competition/CreateCompetition/index.js [hunk 2]
```diff
@@ -26,42 +36,58 @@ export default function CreateCompetitionForm() {
...
             onChange={handleGameInput}
             min={1}
             max={100}
+            required
           />
         </StyledFormLabelInputWrapper>
-        <StyledFormButton>Create competition</StyledFormButton>
+        <StyledButtonComponent
+          type="add"
+          callback={handleSubmit}
```
Comment (actyralf) [inline components/Competition/CreateCompetition/index.js:82]: `callback` is not a good prop name - it's not clear which event triggers this. Is it simply `onClick` or something different?

File: components/GeneralComponents/Buttons/index.js [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+    } else if (type === "back") {
+      return BackIcon;
+    } else if (type === "delete") {
+      return TrashIcon;
+    } else if (type === "archive") {
+      return ArchiveIcon;
+    } else if (type === "add") {
+      return AddIcon;
+    } else if (type === "update") {
+      return SubmitIcon;
```
Comment (actyralf) [inline components/GeneralComponents/Buttons/index.js:38]: You can use an object as a "map" for such things, e.g.:

```js
const buttonIcons{
   edit:  EditIcon,
   back: BackIcon,
   delete: TrashIcon,
   // ...
}
```
you can then access the icon with bracket notation:
```js
   <Image src={buttonIcons[type]} />
```

File: components/GeneralComponents/Buttons/index.js [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+  function fireMessage(intemName) {
+    if (type === "archive") {
+      return ArchiveMessage(intemName);
+    } else if (type === "add") {
+      return SuccessCreateMessage();
+    } else if (type === "update") {
+      return SuccessUpdateMessage(intemName);
+    } else if (type === "delete") {
+      return DeleteMessage();
+    }
```
Comment (actyralf) [inline components/GeneralComponents/Buttons/index.js:51]: same thing as a above: you could change your config map to store the different functions for each type as well.

**Review by actyralf — CHANGES_REQUESTED** [review]
Poooh - far too big for one PR :-).

It looks good overall, I dropped a few comments though...

File: components/Competition/CreateCompetition/index.js (hunk 1 above)
Comment (nikokreissl) [reply components/Competition/CreateCompetition/index.js:30]: In order to use the `StyledButtonComponent` (which comes with the toast message) on forms too I had to pass the button the submitHandleFunction in order to execute it after the timeout.
The form itself is still preventing the default like this `<form onSubmit={(event) => event.preventDefault()}`

It seems to work as intended but is this bad pratice?

File: components/GeneralComponents/Buttons/index.js (hunk 4 above)
Comment (nikokreissl) [reply components/GeneralComponents/Buttons/index.js:51]: Will also take care of it too. Thanks 🙂

**Review by sahedw — APPROVED** [review]
Really good job on implementing the suggestions from the coach. Since the coach already checked your entire code, I'm only able to review the latest commits. Looks good to me!
